R/otp-route.R

```diff
@@ -0,0 +1,287 @@
+#' @param maxWalkDistance Numeric passed to OTP
+#' @param walkReluctance Numeric passed to OTP
+#' @param transferPenalty Numeric passed to OTP
+#' @param minTransferTime Numeric passed to OTP
+#' @param full_elevation Logical, should the full elevation profile be returned, defualt FALSE
+#'
+#' @export
+#'
+#' @detials
+#' This function returns a SF data.frame with one row for each leg of the jounrey
+#' (a leg is defined by a change in mode). For transit more than one route option may be returned
+#' and is indicated by the route_option column.
+#'
+#'
+#' Elevation
+#' OTP supports elevation data, and can return the elevation profile of the route if available.
+#' OTP returns the elevation profile separately from the XY coordinates, this means there is not
+#' direct match between the number of XY points and the number of Z points.  OTP also only returns
+#' the elevation profile for the first leg of the route (this appears to be a bug).
+#' As default the otp_plan function matches the elevation profile to the XY coordinates to return
+#' a SF linestring with XYZ coordinates. If you require a more detailed elevation profile,
+#' the full_elevation parameter will return a nested data.frame with three columns.
+#' first and second are returned from OTP, while distance is the cumulative distance along the
+#' route and is derived from First.
+#'
+otp_plan <- function(otpcon = NA,
+           fromPlace = NA,
+           toPlace = NA,
+           mode = "CAR",
+           date_time = Sys.time(),
+           arriveBy = FALSE,
+           maxWalkDistance = 800,
+           walkReluctance = 2,
+           transferPenalty = 0,
+           minTransferTime = 0,
+           full_elevation = FALSE)
+{
+  # Check Valid Inputs
+  if(!"otpconnect" %in% class(otpcon)){
+    message("otpcon is not a valid otpconnect object")
+    stop()
+  }
+  if(class(fromPlace) != "numeric" | length(fromPlace) != 2){
+    message("fromPlace is not a valid latitude, longitude pair")
+    stop()
+  }else{
+    if(fromPlace[1] <= 90 & fromPlace[1] >= -90 &  fromPlace[2] <= 180 & fromPlace[2] >= -180){
+      fromPlace <- paste(fromPlace, collapse = ",")
+    }else{
+      message("fromPlace coordinates excced valid values +/- 90 and +/- 180 degrees")
+      stop()
+    }
+
+  }
+  if(class(toPlace) != "numeric" | length(toPlace) != 2){
+    message("toPlace is not a valid latitude, longitude pair")
+    stop()
+  }else{
+    if(toPlace[1] <= 90 & toPlace[1] >= -90 &  toPlace[2] <= 180 & toPlace[2] >= -180){
+      toPlace <- paste(toPlace, collapse = ",")
+    }else{
+      message("toPlace coordinates excced valid values +/- 90 and +/- 180 degrees")
+      stop()
+    }
+
+  }
+  if(!all(mode %in% c("TRANSIT","WALK","BICYCLE","CAR","BUS","RAIL"))){
+    message("mode is not a valid, can be a character vector of any of these values TRANSIT, WALK, BICYCLE, CAR, BUS, RAIL")
+    stop()
```

Comment:
Check of mode needs to be a bit more nuanced. Some combinations of these are invalid.

Reply:
uppercase text is easy, what are the rules on which combinations are allowed?

Reply:
I'll probably need to dig into the code to get definite answer on that one, though there may be something in the Google Group. It may be best to stick with the options available in the GUI and the modes they generate in the URL (after clicking the view this itinerary link (or whetever it's called). Leave it with me.